### DISCUSSION TITLE
Custom vertex data

### DISCUSSION THREAD

**Discussion by ghost** [bevyengine/bevy#5408] [Q&A] [answered]
Hey there !
I asked a similar question on bevy's discord where @devil-ira answered that some vertex data (namely positions, normals and UVs) are required when creating a custom mesh. But he also mentioned that this may change for the next release (I assume 0.8 ?).  I'd like to know if it will be possible to create vertices with completely custom data, say, vertices that will only contain position and color ?

**Comment by IceSentry** [comment, marked as answer]
In 0.8 only `position` and `normal` are required attributes. You can add any other attributes you want.

**Reply by ghost** [reply]
Thank you ! And If I wanted to change that, would that be too much work ?

**Reply by IceSentry** [reply]
Shouldn't be too hard, you should start here https://github.com/bevyengine/bevy/blob/1c23421f4141c1e93c120ccfc27ed86edc3488af/crates/bevy_pbr/src/render/mesh.rs#L550

You'll need to add shader_defs for each attributes.

Look at this PR to see how it was done for UVs https://github.com/bevyengine/bevy/pull/5222

**Reply by ghost** [reply]
Thank you again !
